backend/canisters/user/impl/src/model/contacts.rs

```diff
@@ -0,0 +1,73 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use types::{FieldTooLongResult, FieldTooShortResult, OptionUpdate, UserId};
+use user_canister::set_contact::OptionalContact;
+
+const MAX_NICKNAME_LEN: u32 = 32;
+const MIN_NICKNAME_LEN: u32 = 2;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct Contacts {
+    map: HashMap<UserId, Contact>,
+}
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct Contact {
+    pub nickname: Option<String>,
+}
+
+pub enum SetContactResponse {
+    Success,
+    NoChange,
+    NicknameTooShort(FieldTooShortResult),
+    NicknameTooLong(FieldTooLongResult),
+}
+
+impl Contacts {
+    pub fn set_contact(&mut self, contact: OptionalContact) -> SetContactResponse {
+        match contact.nickname {
+            OptionUpdate::NoChange => SetContactResponse::NoChange,
+            OptionUpdate::SetToNone => {
+                if self.map.remove(&contact.user_id).is_some() {
```

Comment:
We need to be careful here that when we add more fields we only remove contacts which are empty.